distributed/shuffle/_worker_extension.py

```diff
@@ -543,12 +542,11 @@ async def get_output_partition(self, i: int, key: str) -> pd.DataFrame:
             data = self._read_from_disk((i,))
 
             def _() -> pd.DataFrame:
-                df = convert_partition(data)
-                return df.to_pandas()
+                return convert_partition(data, self.meta)
 
             out = await self.offload(_)
         except KeyError:
-            out = self.schema.empty_table().to_pandas()
+            out = self.meta.copy()
```

Comment:
Similar question here re: copy

Reply:
I am hesitant to return several references to the same dataframe object for independent partitions since they are not immutable.